sources/mysql/infoschema.go

```diff
@@ -237,38 +228,99 @@ func (isi InfoSchemaImpl) GetColumns(conv *internal.Conv, table common.SchemaAnd
 // other constraints.  Note: we need to preserve ordinal order of
 // columns in primary key constraints.
 // Note that foreign key constraints are handled in getForeignKeys.
-func (isi InfoSchemaImpl) GetConstraints(conv *internal.Conv, table common.SchemaAndName) ([]string, map[string][]string, error) {
-	q := `SELECT k.COLUMN_NAME, t.CONSTRAINT_TYPE
-              FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS AS t
-                INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS k
-                  ON t.CONSTRAINT_NAME = k.CONSTRAINT_NAME AND t.CONSTRAINT_SCHEMA = k.CONSTRAINT_SCHEMA AND t.TABLE_NAME=k.TABLE_NAME
-              WHERE k.TABLE_SCHEMA = ? AND k.TABLE_NAME = ? ORDER BY k.ordinal_position;`
-	rows, err := isi.Db.Query(q, table.Schema, table.Name)
+func (isi InfoSchemaImpl) GetConstraints(conv *internal.Conv, table common.SchemaAndName) ([]string, []schema.CheckConstraint, map[string][]string, error) {
+	tableExists, err := isi.isCheckConstraintsTablePresent()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+
+	finalQuery := isi.getConstraintsDQL(tableExists)
+	rows, err := isi.Db.Query(finalQuery, table.Schema, table.Name)
+	if err != nil {
+		return nil, nil, nil, err
 	}
 	defer rows.Close()
+
 	var primaryKeys []string
-	var col, constraint string
+	var checkKeys []schema.CheckConstraint
 	m := make(map[string][]string)
+
 	for rows.Next() {
-		err := rows.Scan(&col, &constraint)
-		if err != nil {
-			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
-			continue
-		}
-		if col == "" || constraint == "" {
-			conv.Unexpected(fmt.Sprintf("Got empty col or constraint"))
+		if err := isi.processRow(rows, tableExists, conv, &primaryKeys, &checkKeys, m); err != nil {
+			conv.Unexpected(fmt.Sprintf("Can't scan constrants. error: %v", err))
 			continue
 		}
-		switch constraint {
-		case "PRIMARY KEY":
-			primaryKeys = append(primaryKeys, col)
-		default:
-			m[col] = append(m[col], constraint)
-		}
 	}
-	return primaryKeys, m, nil
+
+	return primaryKeys, checkKeys, m, nil
+}
+
+// checkCheckConstraintsTableExists checks if the CHECK_CONSTRAINTS table exists.
+func (isi InfoSchemaImpl) isCheckConstraintsTablePresent() (bool, error) {
```

Comment:
nit: it seems like this method is called only from the getConstraintsDQL method. 

Why not just call it directly from within that method and avoid passing the tableExists Flag ?

Reply:
Added this fix